pkg/node/connection_pool_test.go

```diff
@@ -70,36 +75,34 @@ func TestAdd(t *testing.T) {
 	}
 }
 
-func TestRemove(t *testing.T) {
-
-	conn, err := grpc.Dial("127.0.0.1:0", grpc.WithInsecure())
-	assert.NoError(t, err)
-	// gc.Close = func() error { return nil }
+func TestDial(t *testing.T) {
 	cases := []struct {
-		pool          ConnectionPool
-		key           string
-		expected      interface{}
+		pool          *ConnectionPool
+		node          *pb.Node
 		expectedError error
+		expected      *Conn
 	}{
 		{
-			pool: ConnectionPool{
-				mu:    sync.RWMutex{},
-				items: map[string]*Conn{"foo": &Conn{grpc: conn}},
-			},
-			key:           "foo",
+			pool:          NewConnectionPool(newTestIdentity(t)),
+			node:          &pb.Node{Id: "foo", Address: &pb.NodeAddress{Address: "127.0.0.1:0"}},
 			expected:      nil,
 			expectedError: nil,
 		},
 	}
 
-	for i := range cases {
-		v := &cases[i]
-		err := v.pool.Remove(v.key)
-		assert.Equal(t, v.expectedError, err)
-
-		test, err := v.pool.Get(v.key)
-		assert.Equal(t, v.expectedError, err)
+	for _, v := range cases {
 
-		assert.Equal(t, v.expected, test)
+		go testDial(t, v.pool, v.node, v.expectedError)
+		go testDial(t, v.pool, v.node, v.expectedError)
+		go testDial(t, v.pool, v.node, v.expectedError)
+		go testDial(t, v.pool, v.node, v.expectedError)
 	}
```

Comment:
wait for the dial tests to finish as well

Reply:
done